src/core/Akka/Util/ByteString.cs

```diff
@@ -28,6 +29,46 @@ namespace Akka.IO
     [DebuggerDisplay("(Count = {_count}, Buffers = {_buffers})")]
     public sealed class ByteString : IEquatable<ByteString>, IEnumerable<byte>
     {
+        public class ByteStringReadOnlySequenceSegment : ReadOnlySequenceSegment<byte>
+        {
+            public ByteStringReadOnlySequenceSegment(ReadOnlyMemory<byte> memory, long runningIndex)
+            {
+                Memory = memory;
+                RunningIndex = runningIndex;
+            }
+            
+            /// <remarks>
+            /// This is here because <see cref="ReadOnlySequenceSegment{T}"/>
+            /// has predefined properties with Protected Setters.
+            /// </remarks>
+            public static ReadOnlySequence<byte> CreateSequence(ByteString bs)
+            {
+                var bArr = bs._buffers;
+                var first = new ByteStringReadOnlySequenceSegment(bArr[0],0);
+                ByteBuffer item = default;
+                ByteStringReadOnlySequenceSegment last = first;
+                if (bArr.Length == 2)
+                {
+                    item = bArr[1];
+                    last = new ByteStringReadOnlySequenceSegment(item, bs.Count-item.Count);
+                    first.Next = last;
+                }
```

Comment:
TBH this might be premature optimization....

Reply:
Yes it is, please remove